backend/test/core/usecase/ComputeGroupBalanceUseCase.test.ts

```diff
@@ -0,0 +1,78 @@
+import type { GroupRepository } from "src/core/repository/GroupRepository";
+import { describe, test, expect, beforeEach, vi } from "vitest";
+import { GroupRepositoryMock } from "./test-helpers";
+import { ComputeGroupBalanceUseCase } from "src/core/usecase/ComputeGroupBalanceUseCase";
+import { GroupBalance } from "src/core/entity/GroupBalance";
+import { Group } from "src/core/entity/Group";
+import { GroupMember } from "src/core/entity/GroupMember";
+import { Transaction } from "src/core/entity/Transaction";
+import { computeGroupBalance } from "src/core/behaviour/computeGroupBalance";
+
+describe("Compute balance (use case)", () => {
+  let groupRepo: GroupRepository;
+
+  beforeEach(() => {
+    groupRepo = new GroupRepositoryMock();
+  });
+
+  describe("Happy path", () => {
+    test("Given a group already created with two members and two transactions, it should return the balance of the group", async () => {
+      // Arrange
+      const computeBalance = new ComputeGroupBalanceUseCase(groupRepo);
+      const group = new Group(
+        0,
+        "group name",
+        [
+          new GroupMember(1, "group member 1"),
+          new GroupMember(2, "group member 2"),
+        ],
+        [new Transaction(1, 1, [1, 2], 3), new Transaction(2, 1, [2], 1)]
+      );
+      groupRepo.findById = vi.fn().mockResolvedValue(group);
+      const mock = vi.fn().mockImplementation(computeGroupBalance);
+      mock.mockImplementation(
+        () =>
+          new GroupBalance(
+            0,
+            new Map<number, number>([
+              [1, 2.5],
+              [2, -2.5],
+            ])
+          )
+      );
+
+      // Act
+      const result = await computeBalance.execute(0);
+
+      // Assert
+      const expectedGroupBalance = new GroupBalance(
+        0,
+        new Map<number, number>([
```

Comment:
I would suggest to create the GroupBalance beforehand, and reference it both in the mockImplem and in the expected (more readable)

Reply:
It's just that I thought it was strange to create an expected value outside of the "Assert" part, but why not !